server/interpreter.js

```diff
@@ -943,14 +944,14 @@ Interpreter.prototype.initString = function(scope) {
   this.STRING.addNativeMethod('match', wrapper);
 
   wrapper = function(regexp) {
-    regexp = regexp ? regexp.data : undefined;
+    regexp = regexp ? regexp.regexp : undefined;
     return this.search(regexp);
   };
   this.STRING.addNativeMethod('search', wrapper);
 
-  wrapper = function(substr, newSubStr) {
-    // TODO: Rewrite as a polyfill to support function replacements.
-    return this.replace(substr, newSubStr);
+  wrapper = function(substr, newSubstr) {
+    // Support for function replacements is the responsibility of a polyfill.
+    return String(this).replace(substr.regexp || substr, newSubstr);
```

Comment:
Only until I get round to s/thisInterpreter/intrp/g as we previously discussed...  :-)